### PR TITLE
Fix/OAP-36/LikertScale authoring input states

### PR DESCRIPTION
**Pull request by oatymart**
https://oat-sa.atlassian.net/browse/OAP-36

Fixes an issue where the input controls of the Likert were disabled in Response state, but enabled in Question state.
In the new behaviour, inputs are disabled in both of these states (but appear enabled when the widget is inactive).

The solution is not perfect, because the sequences of callbacks which run can vary:
- **new on canvas**: `afterCreate()` > `initQuestionState` > `render()`
- **existing when Authoring opened**: `initCreator()` > `render()`
- **existing, when changing state**: `render()` > `initQuestionState()` > `exitQuestionState()` etc.

New on canvas (auto Question state) ✔️ 
Screenshot 2024-03-07 at 16 52 07

Answer state entered ✔️ 
Screenshot 2024-03-07 at 16 52 13

Question state entered ✔️ 
Screenshot 2024-03-07 at 16 52 23

Inactive after being active ✔️ 
Screenshot 2024-03-07 at 16 52 31

Initial inactive in re-opened item (could not solve this, as no states are entered)
Screenshot 2024-03-07 at 16 52 44

---

Finally, it also was decided to leave `pci.getResponseDeclaration().setTemplate('NONE');` alone, as not setting it causes the widget to fall into an invalid `correct` state, if xmlEditRp is not installed to provide the `CUSTOM` template option.

### CODE REVIEW

File: migrations/Version202403071639280420_qtiItemPci.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace oat\qtiItemPci\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use Doctrine\Migrations\Exception\IrreversibleMigration;
+use oat\qtiItemPci\model\IMSPciModel;
```
Comment (jsconan) [inline migrations/Version202403071639280420_qtiItemPci.php:10]:
```suggestion
use oat\qtiItemPci\model\PciModel;
```

File: migrations/Version202403071639280420_qtiItemPci.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+final class Version202403071639280420_qtiItemPci extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Install/update PCI LikertScaleInteraction';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $registry = (new IMSPciModel())->getRegistry();
```
Comment (jsconan) [inline migrations/Version202403071639280420_qtiItemPci.php:25]: **issue:** This is not an IMS PCI...
**suggestion:**
```suggestion
        $registry = (new PciModel())->getRegistry();
```

File: migrations/Version202403071639280420_qtiItemPci.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+declare(strict_types=1);
+
+namespace oat\qtiItemPci\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use Doctrine\Migrations\Exception\IrreversibleMigration;
+use oat\qtiItemPci\model\IMSPciModel;
+use oat\qtiItemPci\scripts\install\RegisterPciLikertScaleInteraction;
```
Comment (jsconan) [inline migrations/Version202403071639280420_qtiItemPci.php:11]:
```suggestion
use oat\qtiItemPci\scripts\install\RegisterPciLikertScale;
```

File: migrations/Version202403071639280420_qtiItemPci.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return 'Install/update PCI LikertScaleInteraction';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $registry = (new IMSPciModel())->getRegistry();
+        $this->addReport(
+            $this->propagate(
+                new RegisterPciLikertScaleInteraction()
```
Comment (jsconan) [inline migrations/Version202403071639280420_qtiItemPci.php:28]: **issue:** This class doesn't exist...
**suggestion:**
```suggestion
                new RegisterPciLikertScale()
```

File: migrations/Version202403071639280420_qtiItemPci.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            )(
+                ['0.6.1']
+            )
+        );
+    }
+
+    public function down(Schema $schema): void
+    {
+        throw new IrreversibleMigration(
+            'In order to undo this migration, please revert the client-side changes and run ' . RegisterPciLikertScaleInteraction::class
```
Comment (jsconan) [inline migrations/Version202403071639280420_qtiItemPci.php:38]:
```suggestion
            'In order to undo this migration, please revert the client-side changes and run ' . RegisterPciLikertScale::class
```

**Review by jsconan — CHANGES_REQUESTED** [review]
The fix works. However, I had to correct a few things:
- The migration does not work (classes are incorrect)
- The PCI bundle is not updated

File: migrations/Version202403071639280420_qtiItemPci.php (hunk 2 above)
Comment (oatymart) [reply migrations/Version202403071639280420_qtiItemPci.php]: > This is not an IMS PCI... Did you create it using the SDK?

Guilty.

**Review by jsconan — APPROVED** [review]
All good!

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Pull request merged**